sssom/util.py

```diff
@@ -237,3 +445,275 @@ def sha256sum(filename):
         for n in iter(lambda: f.readinto(mv), 0):
             h.update(mv[:n])
     return h.hexdigest()
+
+def merge_msdf(msdf1:MappingSetDataFrame, msdf2:MappingSetDataFrame, reconcile:bool=True, inplace:bool=False) -> MappingSetDataFrame:
+        """
+        Merging msdf2 into msdf1,
+        if reconcile=True, then dedupe(remove redundant lower confidence mappings) and
+            reconcile (if msdf contains a higher confidence _negative_ mapping,
+            then remove lower confidence positive one. If confidence is the same,
+            prefer HumanCurated. If both HumanCurated, prefer negative mapping).
+
+        Args:
+            msdf1 (MappingSetDataFrame): The primary MappingSetDataFrame
+            msdf2 (MappingSetDataFrame): The secondary MappingSetDataFrame
+            reconcile (bool, optional): [description]. Defaults to True.
+
+        Returns:
+            MappingSetDataFrame: Merged MappingSetDataFrame.
+        """
+
+        merged_msdf = MappingSetDataFrame()
+        # If msdf2 has a DataFrame
+        if msdf2.df is not None:
+            # 'outer' join in pandas == FULL JOIN in SQL
+            merged_msdf.df = msdf1.df.merge(msdf2.df, how='outer')
+        else:
+            merged_msdf.df = msdf1.df
+        #merge the non DataFrame elements
+        merged_msdf.prefixmap = dict_merge(msdf2.prefixmap, msdf1.prefixmap, 'prefixmap')
+        # After a Slack convo with @matentz, commented out below.
+        #merged_msdf.metadata = dict_merge(msdf2.metadata, msdf1.metadata, 'metadata')
```

Comment:
The metadata needs to be added as column values in though? Else we lose it!

Reply:
As per our discussion on Slack or rather as far as I understood from it (and the test example `basic.tsv`), was that metadata was generated from the columns of the `msdf.DataFrame` itself, correct? So they are already in the `msdf.DataFrame`?

Reply:
Done! https://github.com/mapping-commons/sssom-py/pull/72/commits/eb884e4a543da7a0ad872569f2c711922281c8b9